eng/pipelines/templates/stages/archetype-sdk-client.yml

```diff
@@ -24,6 +24,7 @@ stages:
               - Name: Storage Blobs
                 NonWindowsPath: ./build/sdk/samples/storage/blobs/blobs_client_sample
                 WindowsPath: build\sdk\samples\storage\blobs\debug\blobs_client_sample.exe
+                RunCondition: contains(variables['build.args'], '-DTRANSPORT_CURL=ON')
```

Comment:
This is a bit of a stop gap for now. Maybe @danieljurek can come up with a more appropriate/clean solution to run "independent" live tests/samples on different AzDo legs.